reference.py

```diff
@@ -105,7 +105,9 @@ def create_labeled_silent_payment_address(b_scan: ECKey, B_spend: ECPubKey, m: i
 
 
 def decode_silent_payment_address(address: str, hrp: str = "tsp") -> Tuple[ECPubKey, ECPubKey]:
-    version, data = decode(hrp, address)
```

Comment:
in a026b44fa9f6a6e24bda4dd07696694d3aa5af0f

I think we should check a) that version is in the expected range b) fail if version is 31